packages/react/dialog/src/Dialog.tsx

```diff
@@ -164,19 +175,112 @@ const DialogContent = React.forwardRef((props, forwardedRef) => {
   const context = useDialogContext(CONTENT_NAME);
   return (
     <Presence present={forceMount || context.open}>
-      <DialogContentImpl data-state={getState(context.open)} {...contentProps} ref={forwardedRef} />
+      {context.modal ? (
+        <DialogContentModalImpl {...contentProps} ref={forwardedRef} />
+      ) : (
+        <DialogContentNonModalImpl {...contentProps} ref={forwardedRef} />
+      )}
     </Presence>
   );
 }) as DialogContentPrimitive;
 
+DialogContent.displayName = CONTENT_NAME;
+
+type DialogContentImplOwnProps = Omit<
+  Polymorphic.OwnProps<typeof DialogContentPart>,
+  'trapFocus' | 'disableOutsidePointerEvents'
+>;
+
+type DialogContentImplPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof DialogContentPart>,
+  DialogContentImplOwnProps
+>;
+
+const DialogContentModalImpl = React.forwardRef((props, forwardedRef) => {
```

Comment:
Wondering why these aren't just named `DialogContentModal` and `DialogContentNonModal`? I thought we usually only have `Impl` parts when the former is being used and we need it to render a separate part internally.

I went looking for their respective non-impl parts and couldn't find them.

Reply:
I think the `Impl` naming convention we have is a bit general, not always due to `Presence` or similar. That said, I'm very happy to change this here.

Reply:
I usually expect that when there is an `Impl` part that there is an associated non-impl part which is why the impl name had to be used. That's not the case here though which tripped me up. Would be great to change if you wouldn't mind 🙏